src/components/contact/IFXSelectableContact.vue

```diff
@@ -51,68 +90,148 @@ export default {
             item-text="name"
             return-object
-            :rules='formRules.contactable'
           >
           </v-autocomplete>
         </v-col>
       </v-row>
       <v-row>
         <v-col>
-          <v-text-field
-            v-model.trim="itemLocal.contact.name"
-            autocomplete="new-password"
-            :error-messages="errors['contacts.name']"
-            :rules="formRules.generic"
-            label="Name"
-            required
-          ></v-text-field>
-        </v-col>
-        <v-col>
-          <v-select
-            :items='$api.contact.types'
-            v-model.trim='itemLocal.type'
-            autocomplete="new-password"
-            :error-messages="errors['contacts.type']"
-            :rules='formRules.generic'
-            label="Type"
-            required
-          ></v-select>
+          <span v-if="!itemLocal.contact">Select a </span>Contact type
+          <v-radio-group
+            v-model="contactType"
+            row
+          >
+            <v-radio
+              label="Email"
+              value="Email"
+            >
+            </v-radio>
+            <v-radio
+              label="Phone"
+              value="Phone"
+            >
+            </v-radio>
+            <v-radio
+              label="Full"
+              value="Full"
+            >
+            </v-radio>
+          </v-radio-group>
         </v-col>
       </v-row>
-      <v-row>
+      <v-row v-if="contactType === 'Full'">
         <v-col>
-          <v-text-field
-            v-model.trim="itemLocal.contact.detail"
-            autocomplete="new-password"
-            :error-messages="errors['contacts.detail']"
-            :rules="formRules.email"
-            label="Email"
-            required
-          ></v-text-field>
+          <v-row>
+            <v-col>
+              <v-text-field
+                v-model.trim="itemLocal.role"
+                autocomplete="new-password"
+                :error-messages="errors['contacts.role']"
+                :rules="formRules.generic"
+                label="Role"
+                required
+                @input="checkValidForm()"
+              ></v-text-field>
+            </v-col>
+          </v-row>
+          <v-row>
+            <v-col>
+              <v-text-field
+                v-model.trim="itemLocal.detail"
+                autocomplete="new-password"
+                :error-messages="errors['contacts.detail']"
+                :rules="formRules.email"
+                label="Email"
+                required
+                @input="checkValidForm()"
+              ></v-text-field>
+            </v-col>
+            <v-col>
+              <v-text-field
+                v-model.trim="itemLocal.phone"
+                autocomplete="new-password"
+                :error-messages="errors['contacts.phone']"
+                label="Phone"
+                @input="checkValidForm()"
+              ></v-text-field>
```

Comment:
Do you want `formRules.phone` here? And, perhaps, `data-cy` attributes for setting these?

Reply:
Not really.  We talked about phone rules, but they are too complicated if you allow international.